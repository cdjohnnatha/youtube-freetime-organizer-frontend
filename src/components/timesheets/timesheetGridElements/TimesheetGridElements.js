import React from 'react'
import PropTypes from 'prop-types'
import { Grid, Typography } from '@material-ui/core'

const TimesheetGridElements = ({ title, content, direction }) => (
  <Grid item style={{ marginBottom: '1em' }}>
    <Grid container item direction={direction}>
      <Typography component="h1" style={{ marginRight: '1em' }}>
        {title}
      </Typography>
      <Typography
        variant="subtitle1"
        style={{ color: 'grey' }}
      >
        {content}
      </Typography>
    </Grid>
  </Grid>
);

TimesheetGridElements.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
  direction: PropTypes.string,
};

TimesheetGridElements.defaultProps = {
  title: '',
  content: '',
  direction: 'row',
}

export default TimesheetGridElements;
