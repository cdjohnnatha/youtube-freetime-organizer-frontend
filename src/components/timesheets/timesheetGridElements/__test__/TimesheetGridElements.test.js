
import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import '../../../../../jest.setup';
import TimesheetGridElements from '../TimesheetGridElements';

import titleContentMock from './__mock__/titleContent.mock.json';

let wrapper = null;
describe('Test TimesheetGridElements component', () => {
  test('TimesheetGridElements snapshot renders', () => {
    const component = create(<TimesheetGridElements content={titleContentMock.content} title={titleContentMock.title} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  
  test('TimesheetGridElements component', () => {
    wrapper = mount(<TimesheetGridElements content={titleContentMock.content} title={titleContentMock.title} />);
    expect(wrapper.find('h1').text()).toBe(titleContentMock.title)
    expect(wrapper.find('h6').text()).toBe(titleContentMock.content)
  });
});