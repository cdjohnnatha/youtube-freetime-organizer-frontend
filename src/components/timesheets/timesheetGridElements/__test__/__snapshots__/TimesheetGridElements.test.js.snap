// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test TimesheetGridElements component TimesheetGridElements snapshot renders 1`] = `
<div
  className="MuiGrid-root MuiGrid-item"
  style={
    Object {
      "marginBottom": "1em",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-item"
  >
    <h1
      className="MuiTypography-root MuiTypography-body1"
      style={
        Object {
          "marginRight": "1em",
        }
      }
    >
      Test
    </h1>
    <h6
      className="MuiTypography-root MuiTypography-subtitle1"
      style={
        Object {
          "color": "grey",
        }
      }
    >
      Content
    </h6>
  </div>
</div>
`;
